[CS2103T-T13-4] Mark

Na-Nazhou opened the pull request:
@dorcastan @openorclose @kxyoke @TSAI-HSIAO-HAN

rongrongrr commented on docs/DeveloperGuide.adoc:
```diff
@@ -277,53 +279,221 @@ Refer to the guide <<DevOps#, here>>.
+
+1.  User requests to list bookmarks
+2.  Mark shows a list of bookmarks
++
```
Extension for 2. e.g. if there are no bookmarks

rongrongrr reviewed (commented):
Hey! Great job, we see that you guys have started morphing your project. It's great to see that you are on track :)

openorclose committed:
Improve folder ui

kxyoke committed:
Update UG annotations feature

openorclose committed:
Merge pull request #252 from openorclose/folder-ui

Improve folder ui

TSAI-HSIAO-HAN committed:
change style

dorcastan committed:
Fix: Rearrange UG Features and add Mark overview section

Arranged commands into sections according to the type of data that they
operate on. Added an overview of Mark's layout in a top-down manner.

TSAI-HSIAO-HAN committed:
edit data

TSAI-HSIAO-HAN committed:
formatting

dorcastan committed:
Fix: Incorrect heading levels

dorcastan committed:
Fix: Apply suggestions from code review

Amended the walk-through description for AddCommand and re-included
text details for other examples, as suggested by @kxyoke.

TSAI-HSIAO-HAN committed:
exit

TSAI-HSIAO-HAN committed:
fix error

TSAI-HSIAO-HAN committed:
fix

kxyoke committed:
Merge pull request #249 from kxyoke/user-guide

Update User guide to mark out areas that need screenshots

kxyoke committed:
Commit suggestions for ug

kxyoke committed:
Merge branch 'master' into update-docs

TSAI-HSIAO-HAN committed:
merge upstream

TSAI-HSIAO-HAN committed:
formatting

kxyoke committed:
Remove offline page from web browser so any failures lead to about:blank. Logger message regarding internet connection is updated accordingly. Users will have to check themselves if their input was wrong or no internet connection

TSAI-HSIAO-HAN committed:
add test

kxyoke committed:
Remove commented-out code

dorcastan committed:
Merge pull request #248 from dorcastan/update-docs

Reformat User Guide

TSAI-HSIAO-HAN committed:
add css reminder-cell class

TSAI-HSIAO-HAN committed:
formatting

kxyoke committed:
Conform to checkstyle

dorcastan committed:
Update: UG autotag section + minor format fixes

Team tasks: Add overview of Features section and move command
parameters table to Command Summary. Insert horizontal line to separate
Features subsections.

Individual tasks: Update Autotags section and link PPP to the correct
UG section.

dorcastan committed:
Update: Libraries used (README.md)

Add JSoup and Readability4J.
Addresses #57.

TSAI-HSIAO-HAN committed:
fix timer problem

kxyoke committed:
Delete fatal error log

TSAI-HSIAO-HAN committed:
Merge pull request #251 from TSAI-HSIAO-HAN/goto-reminder

Goto reminder command

TSAI-HSIAO-HAN committed:
Merge branch 'master' into improvement

TSAI-HSIAO-HAN committed:
improve notification

TSAI-HSIAO-HAN committed:
delete println

kxyoke committed:
Merge pull request #253 from kxyoke/embed-browser

Update browser address parsing to just load about:blank

TSAI-HSIAO-HAN committed:
fix ui

dorcastan committed:
Fix: UI resizing issues in Dashboard panel

Use AnchorPanes to fix the Label size. Bind the prefWidthProperty
of reminderList to widthProperty of the scrollPane so that it resizes
together.

Also move CSS styling out from FXML to DarkTheme.css

TSAI-HSIAO-HAN committed:
Merge pull request #1 from dorcastan/reminder-list-ui

Fix: UI resizing issues in Dashboard panel

TSAI-HSIAO-HAN committed:
+EOF

dorcastan committed:
Update: Class diagrams (DevGuide Model component)

Separated the giant class diagram into packages so it's easier to read.
Inserted diagrams into DevGuide with a rough text description.

Other minor changes: Removed random newline in Mark.java imports,
edited PPP to reflect number of sections included.

dorcastan committed:
Fix: Remove johndoe.adoc

Updated the link to KX's PPP in AboutUs.

TSAI-HSIAO-HAN committed:
Merge pull request #228 from TSAI-HSIAO-HAN/improvement

Reminder Notification

kxyoke committed:
Add major enhancement annotation to ppp

openorclose committed:
Add folder-add, folder-delete

TSAI-HSIAO-HAN committed:
UG

TSAI-HSIAO-HAN committed:
Merge branch 'master' into improvement

Na-Nazhou committed:
Improve UI

Na-Nazhou committed:
Merge pull request #256 from Na-Nazhou/nazhou/doc

Standardise the styles of favorites panel

kxyoke committed:
Change sd for DG and do slight refactor for annotation commands

kxyoke committed:
Modify add annotation sd

dorcastan committed:
Add: EditAutotagCommand

Refactor AutotagController#removeTagger(String) to return an Optional
instead of a boolean value. No tests yet.

TSAI-HSIAO-HAN committed:
DG

kxyoke committed:
Edit dg delete annotations feature

TSAI-HSIAO-HAN committed:
ppp

kxyoke committed:
Add sd for edit annotation feature in dg

dorcastan committed:
Add: Tests for EditAutotagCommand

And improve coverage for existing tests.

dorcastan committed:
Fix: Typos in JavaDocs

Remove references to AutotagCommand and deleting autotags.

dorcastan committed:
Fix: Phrasing of UG and command summary

Avoid usage of command sentences. Add goto-reminder to command summary.

dorcastan committed:
Add: UserGuide description of autotag-edit

dorcastan committed:
Fix: Error message for autotag-edit

Add a custom message when no fields are changed, in case the user
inputs a new tag name that is the same as the original, with no new
conditions. (Previously, MESSAGE_USAGE was shown.)

dorcastan committed:
Fix: Checkstyle errors

TSAI-HSIAO-HAN committed:
delete space

kxyoke committed:
Frame dg for annotation design considerations

openorclose committed:
Update ppp

openorclose committed:
Apply code suggestions

openorclose committed:
Add folderstructure tests

kxyoke committed:
Edit design considerations for annotations

Na-Nazhou committed:
Add documentation

kxyoke committed:
Touch up annotation feature of dg

kxyoke committed:
Update annotation cd

kxyoke committed:
Update annotation structure of dg

dorcastan committed:
Apply suggestions from code review

Co-Authored-By: Open O. Close <3646725+openorclose@users.noreply.github.com>

kxyoke committed:
Add test case for dg

kxyoke committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T13-4/main into ppp

openorclose committed:
Update UGDG

openorclose committed:
Update link

openorclose committed:
Update ppp

openorclose committed:
Minor refactor

TSAI-HSIAO-HAN committed:
Merge pull request #254 from dorcastan/update-docs

Update documentation — UG Autotags and DG design of Model

openorclose committed:
Merge remote-tracking branch 'origin/master' into folder-commands

# Conflicts:
#	docs/UserGuide.adoc

openorclose committed:
Add folder-edit, -delete, command summary

openorclose committed:
Apply suggestions from code review

Co-Authored-By: dorcastan <46101221+dorcastan@users.noreply.github.com>

openorclose committed:
Fix checkstyle

Na-Nazhou committed:
Add instructions for manual testing

openorclose committed:
Merge remote-tracking branch 'origin/master' into ppp2

# Conflicts:
#	docs/UserGuide.adoc

openorclose committed:
Fix typo

openorclose committed:
Fix ppp

kxyoke committed:
Edit ppp

kxyoke committed:
Make small updates to ppp

kxyoke committed:
Add one liner justification for design chosen

TSAI-HSIAO-HAN committed:
ppp

kxyoke committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T13-4/main into developers-guide

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/images/AnnotationPackageClassDiagram.png

TSAI-HSIAO-HAN committed:
Merge pull request #255 from openorclose/folder-commands

Add folder-edit, folder-delete

dorcastan committed:
Refactor: EditAutotagCommandParser to simplify #parse()

Move validation into AutotagCommandParser#makePredicateFromMultimap()
and call it in EditAutotagCommandParser#parse(args).

dorcastan committed:
Merge branch 'master' into improve-autotag

# Conflicts:
#	docs/UserGuide.adoc

TSAI-HSIAO-HAN committed:
edit ppp

TSAI-HSIAO-HAN committed:
test

TSAI-HSIAO-HAN committed:
Merge branch 'master' into improvement

# Conflicts:
#	docs/UserGuide.adoc
#	docs/diagrams/ModelClassDiagram.puml
#	docs/images/ModelClassDiagram.png

TSAI-HSIAO-HAN committed:
edit ug

Na-Nazhou committed:
Improve user guide

TSAI-HSIAO-HAN committed:
manually test case

Na-Nazhou committed:
Edit developer guide

Na-Nazhou committed:
Add more instructions for manual testing

Na-Nazhou committed:
Add more instructions for manual testing

dorcastan committed:
Merge pull request #258 from dorcastan/improve-autotag

Add autotag-edit

openorclose committed:
Merge remote-tracking branch 'upstream/master' into pppdg

openorclose committed:
Add examples

TSAI-HSIAO-HAN committed:
Apply suggestions from code review

Co-Authored-By: Open O. Close <3646725+openorclose@users.noreply.github.com>

TSAI-HSIAO-HAN committed:
Update docs/diagrams/ReminderAssociationDiagram.puml

dorcastan committed:
Update: Model section of DG

Update class diagrams and add a better description paragraph for each
package.

dorcastan committed:
Fix: UML notation in autotag package class diagram

Arrow should have a dotted line, not a solid line.

TSAI-HSIAO-HAN committed:
edit UG

TSAI-HSIAO-HAN committed:
add ui

TSAI-HSIAO-HAN committed:
Merge pull request #260 from Na-Nazhou/doc

Add ppp, manual testing instructions and edit ug dg

TSAI-HSIAO-HAN committed:
Merge pull request #259 from kxyoke/developers-guide

Update developers guide - annotation feature and model's annotation

TSAI-HSIAO-HAN committed:
Merge pull request #261 from openorclose/pppdg

Add PPP(Daryl) update ug and dg for folders

TSAI-HSIAO-HAN committed:
edit ppp

TSAI-HSIAO-HAN committed:
Merge branch 'master' into improvement

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/images/ModelClassDiagram.png

TSAI-HSIAO-HAN committed:
fix model puml

openorclose committed:
update ppp

TSAI-HSIAO-HAN committed:
add pdf

TSAI-HSIAO-HAN committed:
rename

TSAI-HSIAO-HAN committed:
remove pdf

TSAI-HSIAO-HAN committed:
Merge pull request #262 from openorclose/update-ppp

update ppp

dorcastan committed:
Update: Autotag implementation in DevGuide

Update object and sequence diagrams. Add better description.

Na-Nazhou committed:
Fix ppp

TSAI-HSIAO-HAN committed:
Merge pull request #257 from TSAI-HSIAO-HAN/improvement

Edit UG, DG, PPP

openorclose committed:
update ppp again

Na-Nazhou committed:
Add section numbers for ppp

dorcastan committed:
Merge branch 'master' into update-docs

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/diagrams/AnnotationPackageClassDiagram.puml
#	docs/diagrams/ModelClassDiagram.puml
#	docs/images/AnnotationPackageClassDiagram.png
#	docs/images/ModelClassDiagram.png

openorclose committed:
save

openorclose committed:
move section

TSAI-HSIAO-HAN committed:
add section num

dorcastan committed:
Fix: Reorganise manual testing instructions section

To make it easier to follow

openorclose committed:
Update Storage component

TSAI-HSIAO-HAN committed:
change title

openorclose committed:
Merge pull request #265 from openorclose/update-ppp-again

update ppp again

dorcastan committed:
Update docs/diagrams/ModelClassDiagram.puml

Co-Authored-By: Na Nazhou <42093424+Na-Nazhou@users.noreply.github.com>

TSAI-HSIAO-HAN committed:
Merge pull request #263 from Na-Nazhou/doc

Fix ppp(Nazhou)

dorcastan committed:
Update docs/DeveloperGuide.adoc

TSAI-HSIAO-HAN committed:
Apply suggestions from code review

Co-Authored-By: dorcastan <46101221+dorcastan@users.noreply.github.com>

TSAI-HSIAO-HAN committed:
Apply suggestions from code review

Co-Authored-By: dorcastan <46101221+dorcastan@users.noreply.github.com>

TSAI-HSIAO-HAN committed:
Apply suggestions from code review

TSAI-HSIAO-HAN committed:
Update docs/team/TSAI-HSIAO-HAN.adoc

TSAI-HSIAO-HAN committed:
Merge pull request #267 from openorclose/storage-dev

update dg storage component

TSAI-HSIAO-HAN committed:
Merge pull request #264 from dorcastan/update-docs

Update Developer Guide (Model component and Autotag implementation)

dorcastan committed:
Fix: AutotagController#hasTagger() compares predicates

Only tagger names (tag names) should be compared.

TSAI-HSIAO-HAN committed:
Merge pull request #266 from TSAI-HSIAO-HAN/improvement

Edit ppp

dorcastan committed:
Add: Manual testing instructions for autotags

dorcastan committed:
Fix: Missing import folder during import

Create the folder ImportedBookmarks if it does not already exist when
importing bookmarks.

dorcastan committed:
Add: Manual testing instructions for export/import

Also, fixed some details for Annotation package in the description of
the Model component.

dorcastan committed:
Fix: Failing tests

The previous fix broke ImportCommandTest (needed to add IMPORT_FOLDER
to expectedMark when importing).

kxyoke committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T13-4/main into ppp

kxyoke committed:
Configure gh-pages to not show url

kxyoke committed:
Update code contributed links

kxyoke committed:
Update includes for ppp

kxyoke committed:
Try includes

kxyoke committed:
Update ppp

kxyoke committed:
Update dg test case phrasing

kxyoke committed:
Update ppp content

openorclose committed:
Update Storage component

TSAI-HSIAO-HAN committed:
edit ppp

dorcastan committed:
Merge pull request #268 from dorcastan/update-dev-guide

Add testing instructions (autotags, export, import) + Fix autotag & import bugs

kxyoke committed:
Add ug screenshot for annotate

TSAI-HSIAO-HAN committed:
edit ppp

dorcastan committed:
Update: Autotag implementation in DevGuide

openorclose committed:
Add screenshots

openorclose committed:
Merge pull request #270 from openorclose/daryl-ppp

update ppp

openorclose committed:
Add screenshots

kxyoke committed:
Update dashboard pic

kxyoke committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T13-4/main into user-guide

kxyoke committed:
Add ui screenshots

openorclose committed:
Update screenshots

openorclose committed:
Add param constraint

kxyoke committed:
Update reference to cd for annotation

dorcastan committed:
Update: Project Portfolio Page

Now to cut it down to 10 pages.

kxyoke committed:
Make minor edits to contributions

kxyoke committed:
Cut annotation feature description

openorclose committed:
fix punctuation

kxyoke committed:
Update cross doc link in dg

kxyoke committed:
Cut ppp

kxyoke committed:
Test

kxyoke committed:
Try to fix ppp include problem

dorcastan committed:
Fix: Make PPP shorter

dorcastan committed:
Merge pull request #272 from dorcastan/update-docs

Update: Autotag implementation in DevGuide

dorcastan committed:
Fix: Simplify autotag state diagrams

No need to illustrate non-tagged bookmarks (in the interest of saving
space).

dorcastan committed:
Merge pull request #275 from kxyoke/user-guide

Update images for User guide

dorcastan committed:
Fix: No newline at EOF

openorclose committed:
Merge pull request #274 from openorclose/update-images

Update images

dorcastan committed:
Fix: Number of levels displayed in UG TOC

Table of Contents should show commands too.

Na-Nazhou committed:
Mark ui screenshots

dorcastan committed:
Merge pull request #271 from TSAI-HSIAO-HAN/improvement

edit ppp

kxyoke committed:
Cut dg significantly

Na-Nazhou committed:
Add favorite ui screenshot

kxyoke committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T13-4/main into ppp

# Conflicts:
#	docs/UserGuide.adoc
#	docs/images/ui-screenshots/AddAnnotationCommandUi1.png
#	docs/images/ui-screenshots/AddAnnotationCommandUi2.png

openorclose committed:
Add favorite image

dorcastan committed:
Update: Import and Export commands (UG)

And make minor tweaks to PPP.

openorclose committed:
Fix sequence diagram

kxyoke committed:
Edit screenshots for ug

openorclose committed:
Merge pull request #278 from openorclose/up-seq

Fix sequence diagram

Na-Nazhou committed:
Merge pull request #277 from Na-Nazhou/doc

Add ui screenshots for favorite command

kxyoke committed:
Insert toc to ppp

Na-Nazhou committed:
Rename images

dorcastan committed:
Add: Export and Import command screenshots (UG)

Courtesy of @openorclose.

dorcastan committed:
Merge branch 'master' into update-docs

kxyoke committed:
Edit ppp content

dorcastan committed:
Update: Autotag command screenshots

Again, @openorclose's work.

kxyoke committed:
Remove toc due to insufficient space

Na-Nazhou committed:
Increase the size of ui.png

dorcastan committed:
Update: Command screenshots

Highlight changes

kxyoke committed:
Make minor changes

openorclose committed:
Fix images

Na-Nazhou committed:
Merge pull request #279 from Na-Nazhou/fix-images

Rename images

openorclose committed:
Merge pull request #280 from openorclose/fix-images

Fix images

kxyoke committed:
Reformat ppp slightly

Na-Nazhou committed:
Fix ui.png on readme

kxyoke committed:
Make final changes

openorclose committed:
Fix more images

kxyoke committed:
Format line

openorclose committed:
Merge pull request #282 from openorclose/pics

Fix more images

kxyoke committed:
Make figure a bit smaller.

Na-Nazhou committed:
Increase the size for add command examples

openorclose committed:
Highlight images

dorcastan committed:
Update: PPP (final)

kxyoke committed:
Merge pull request #281 from Na-Nazhou/fix-images

Fix ui.png in readme

dorcastan committed:
Update: Autotag command screenshots

dorcastan committed:
Update: UG (final)

Unless screenshots need to be resized.

openorclose committed:
update rminder

dorcastan committed:
Merge branch 'master' into update-docs

TSAI-HSIAO-HAN committed:
Merge pull request #284 from openorclose/hi

Hi

TSAI-HSIAO-HAN committed:
update ug

dorcastan committed:
Merge pull request #276 from dorcastan/update-docs

Update PPP

dorcastan committed:
Update: statements to sync with PPP submission

openorclose committed:
Merge pull request #269 from kxyoke/ppp

Update ppp to fit within 10 pages

TSAI-HSIAO-HAN committed:
Merge branch 'master' into improvement

# Conflicts:
#	docs/UserGuide.adoc

TSAI-HSIAO-HAN committed:
highlight

dorcastan committed:
Merge pull request #285 from dorcastan/update-docs

Tiny change to PPP

openorclose committed:
fix

openorclose committed:
Merge pull request #286 from TSAI-HSIAO-HAN/improvement

Highlight

openorclose committed:
fix

dorcastan committed:
Fix: Typos and missing testing instructions

openorclose committed:
fix

dorcastan committed:
Fix: Missing tab command summary

TSAI-HSIAO-HAN committed:
ug

openorclose committed:
Merge pull request #287 from openorclose/dd

fix

dorcastan committed:
Merge pull request #288 from dorcastan/fixes

Fix: Typos and missing testing instructions

TSAI-HSIAO-HAN committed:
remove

TSAI-HSIAO-HAN committed:
Merge pull request #289 from TSAI-HSIAO-HAN/improvement

ug

kxyoke committed:
Fix delete all annotations bug: handle case when no cache is available.

TSAI-HSIAO-HAN committed:
change ppp

kxyoke committed:
Merge pull request #298 from TSAI-HSIAO-HAN/improvement

change ppp

TSAI-HSIAO-HAN committed:
Merge pull request #301 from kxyoke/master

Fix delete all annotations bug: handle case when no cache is available.
